tests/ui/derive.rs

```diff
@@ -130,3 +130,16 @@ fn issue14558() {
 }
 
 fn main() {}
+
```

Comment:
Could you add a deny unfulfilled_expectations so that no future commits allows it and then another one breaks this again?

Reply:
Done.

Reply:
Note that an alternative would be the second commit of #15723. If you want, I can revert the change I made here, and clean up #15723 to only add the option to `Clippy.toml`. Otherwise, I'll submit #15723 for discussion after this one is merged.

Reply:
From the discussion on #15723, it appears that `-Dwarnings` is used, so the `#![deny(unfulfilled_lint_expectations)]` is not needed as it transformed from a warning (its default state) into an error.

I'll revert the changes.

Reply:
:+1: